k-distribution/tests/regression-new/map-symbolic-tests/lookup-7-spec.k.out

```diff
@@ -0,0 +1,4 @@
+<k>
+  lookupResult ( X |-> 1
+  ( MAP [ Y <- 2 ] ) [ Z ] )
+</k>
```

Comment:
`(X |-> 1 MAP) [Z]` is the fully evaluated term.

this should be obtained simply by adding the side condition to the rule, as suggested earlier:
```
rule ((K1 |-> V1) MAP) [ K2 <- V2 ] => (K1 |-> V1) (MAP [ K2 <- V2 ]) requires K1 =/=K K2
```

Reply:
I've added tests which exercise both `in_keys` and `notBool in_keys` cases.